### PR TITLE
Hunstman

### PR DESCRIPTION
**Pull request by evakill**

### CODE REVIEW

File: server/studyspaces.py [hunk 1]
```diff
@@ -9,37 +9,16 @@
 from penn.base import APIError
 from .models import StudySpacesBooking, User
 from .penndata import studyspaces
+from .penndata import wharton
 from .base import cached_route
 
+def get_sessionid():
+    return "l87ugy7ha9fabhxil71zwshbsl9ogl7x"
```
Comment (ezwang) [inline server/studyspaces.py]: Don't put any sensitive variables in source code. The repository is public, so everyone can see them.

Instead, you want to use `os.environ` and read the sensitive value from an environment variable. You can then later set this environment variable on the server when you deploy to production.

File: server/studyspaces.py [hunk 2]
```diff
@@ -59,32 +38,34 @@ def parse_times(building):
...
-                    time["end"] = time["to"]
-                    del time["from"]
-                    del time["to"]
-                rooms["rooms"].append(room)
-    except APIError as e:
-        return jsonify({"error": str(e)})
-
+    if building == 1:
+        sessionid = get_sessionid();
+        print(sessionid)
```
Comment (ezwang) [inline server/studyspaces.py]: Remove semicolon and print statements before merging. You can use `flake8` to check for style issues before pushing.

**Comment by joshdoman** [comment]
@evakill Looks really good. It doesn't seem like you can ask for different days in /availability/1 like you can in /gsr. Is that correct? If so, that's one small thing we need to add.

File: server/studyspaces.py [hunk 3]
```diff
@@ -113,32 +106,35 @@ def parse_times(building):
...
-                    del time["to"]
-                rooms["rooms"].append(room)
-    except APIError as e:
-        return jsonify({"error": str(e)})
-
+    if building == 1:
+        sessionid = get_wharton_sessionid(public=True)
+        rooms = wharton.get_wharton_gsrs(sessionid, date=start)
+        rooms = wharton.switch_format(rooms)
+        save_wharton_sessionid()
```
Comment (joshdoman) [inline server/studyspaces.py]: We should only save the Session ID if there is not an error in the response.

File: server/studyspaces.py [hunk 4]
```diff
@@ -251,21 +248,37 @@ def book_room():
...
         user = None
 
-    resp = studyspaces.book_room(room, start.isoformat(), end.isoformat(), **contact)
-    if "results" in resp:
+    if lid == 1:
+        sessionid = get_wharton_sessionid()
+        if not sessionid:
+            return jsonify({"results": False, "error": "You must pass a sessionid when booking a Wharton GSR!"})
+        resp = wharton.book_reservation(sessionid, room, start, end)
+        save_wharton_sessionid()
```
Comment (joshdoman) [inline server/studyspaces.py]: We should only save the session id if the booking goes through. That way we know it's valid.

**Pull request merged**